### ISSUE TITLE
Web Components with htmx 2.0.0-alpha1 - Fields not detected on submit

### ISSUE THREAD

**Issue by fredt34** [bigskysoftware/htmx#2285]
I'm trying to use Web Components - here, Shoelace - with htmx v2.

If I read the docs correcly, this should post firstName and lastName on Submit?

```
<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <title>Test</title>
  <!--script src="https://unpkg.com/htmx.org@1.9.3"></script-->
  <script src="https://unpkg.com/htmx.org@2.0.0-alpha1/dist/htmx.min.js"></script>
  <script src="https://unpkg.com/hyperscript.org@0.9.9"></script>
  <!-- Shoelace -->
  <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@shoelace-style/shoelace@2.13.1/cdn/themes/light.css" />
  <script type="module" src="https://cdn.jsdelivr.net/npm/@shoelace-style/shoelace@2.13.1/cdn/shoelace.js" ></script>
</head>

<body>
  <div>
      <form>
          <sl-input name="firstName" id="first" label="What is your First Name?" > 
          </sl-input>
          <div class="form-group">
              <label>Last Name</label>
              <input type="text" class="form-control" name="lastName">
          </div>
          <sl-button hx-post="/update" hx-target="#target">Submit</sl-button>
      </form>
      <div id="target" style="max-width:1024px"></div>
  </div>
</body>
```

But according to the Debugger Network tab in Chrome only lastName is transmitted.

It's highly possible that my code is awfully wrong...

I'd appreciate any tip or fix.

**Comment by fernandojpps** [comment]
@fredt34 

TLDR: add `hx-include="sl-input"` to your `sl-button`

I would also suggest that you move the `hx-post`, `hx-target` and `hx-include` to the `form` and add a `type="submit"` to the button. This helps ensure that all ways of form submission are handled via HTMX. At the moment if a user presses the enter key, the form will be submitted the "old way".

---

Why do you need to add `hx-include`?
Because HTMX is unable to get the `input` since it lives inside the shadow-dom. By adding `hx-include="sl-input"`, you are telling HTMX that you want to use the values from all `sl-input` elements. You could also use `hx-include="[name='firstName']"`, but you would have to manually add all `sl-input` elements.

I don't think HTMX could support this without a significant performance penalty. Since [it's not possible to use a single selector anymore to get elements inside the shadow-dom](https://developer.chrome.com/blog/remove-shadow-piercing?hl=pt-br), HTMX would need to check every single element for a `shadowRoot`, where it could `querySelect` elements inside that particular shadow root.

I hope this helps

**Comment by fredt34** [comment]
And It Just Worked
Thanks a LOT
